### PR TITLE
Add dns check to domains

### PR DESCRIPTION
**Pull request by nickspaargaren**
This PR adds a pipeline that lists the domains that don't have an A record. 

ELI5: Google domains that don't exist anymore should be listed or automatically removed in by Github Action. With this change we can remove unused domains to make the list more compact.

### How to test
1. Run `make` to see the new `make dnscheck` command.
2. Run the new `make dnscheck` command.
3. Make sure all domains without a response get automaticcly removed.

### CODE REVIEW

**Comment by touhidurrr** [comment]
@nickspaargaren I do not like this approach. Only `dns.resolver.NXDOMAIN` errors should be  honored. Other errors might be temporary.

**Comment by Bundy01** [comment]
Yes, see https://github.com/nickspaargaren/no-google/issues/203

**Comment by nickspaargaren** [comment]
@touhidurrr, you mean like [this](https://github.com/nickspaargaren/no-google/pull/189/commits/80a1fe2210fae3a472040695c18f3266865199c5)?

File: scripts/dnscheck.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import dns.resolver
+
+def check_domain(domain):
+    try:
+        # Query for 'A' records
+        answers = dns.resolver.resolve(domain, 'A')
+        # If we get any answers, the domain has a response
+        return bool(answers)
+    except (dns.resolver.NoAnswer, dns.resolver.NXDOMAIN, dns.resolver.Timeout):
```
Comment (touhidurrr) [inline scripts/dnscheck.py:9]: Only NXDOMAIN is enough. Definitely not timeout as it is a DNS server issue. No Answer has a lot of weird cases so I don't know when it should be honored.

We should prevent losing active domains in this checkup. Thus again, I think NXDOMAIN is enough.

Comment (nickspaargaren) [reply scripts/dnscheck.py]: Thanks, done!

File: scripts/dnscheck.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        has_a_record = True
+    except (dns.resolver.NXDOMAIN, Exception):
+        has_a_record = False
+    
+    try:
+        # Query for 'NS' records
+        dns.resolver.resolve(domain, "NS")
+        has_ns_record = True
+    except (dns.resolver.NXDOMAIN, Exception):
+        has_ns_record = False
```
Comment (touhidurrr) [inline scripts/dnscheck.py:17]: maybe do it like this? so that we only remove a domain if we are sure that it does not exist anymore? otherwise not.

Something like this:
```py3
try: 
  dns.resolver.resolve(domain, "NS")
  has_ns_record = True
except dns.resolver.NXDOMAIN:
  has_ns_record = False
except:
  has_ns_record = True
```

**Review by touhidurrr — APPROVED** [review]

File: scripts/dnscheck.py (hunk 2 above)
Comment (nickspaargaren) [reply scripts/dnscheck.py:10]: Done, can you check the latest changes? Domains now get automatically removed as well!
